### PR TITLE
ENH allow choose kpis when generate budget control

### PR DESCRIPTION
**Pull request by kittiu**
The concept of this PR is, now we can store the KPIs in budget_contol. So, there is no need to pass kpis by context. 

Note: 
* we will sue "Reset Plan" as normal, but it will always looks into tab "Reset Options" to know what KPIs to use. I will use this tab also for Job Order options.
* budget_control_select_plan is removed.

image

The wizard will just passing "kpi_ids" to all budget_control. And "Reset Plan" will use it to create lines.

image

**Please check if this affect budget_plan_revision**

### CODE REVIEW

File: budget_plan_revision/models/budget_control.py [hunk 1]
```diff
@@ -20,27 +20,28 @@ class BudgetControl(models.Model):
...
-                )
-            )
-            if kpi:
-                kpi_id = kpi[0].get(analytic_id)
-            ctx["kpi_ids"] = kpi_id
-        domain_kpi = super(
-            BudgetControl, self.with_context(ctx)
-        )._domain_kpi_expression()
-        return domain_kpi
+    # We shouldn't need this anymore???
```
Comment (kittiu) [inline budget_plan_revision/models/budget_control.py]: I comment this out, I am not sure it is still needed.

File: budget_control_select_plan/views/mis_report_kpi.xml [hunk 2]
```diff
@@ -1,16 +0,0 @@
-<?xml version="1.0" encoding="utf-8" ?>
-<!-- Copyright 2021 Ecosoft Co., Ltd (https://ecosoft.co.th)
-     License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html) -->
-<odoo>
-    <record id="mis_report_view_kpi_tree" model="ir.ui.view">
-        <field name="name">mis.report.view.kpi.tree</field>
-        <field name="model">mis.report.kpi</field>
-        <field name="arch" type="xml">
-            <tree>
```
Comment (Saran440) [inline budget_control_select_plan/views/mis_report_kpi.xml:9]: you should add tree view on mis.report.kpi for user friendly

**Comment by Saran440** [comment]
Budget Revision can't reset plan
Step to error:
1. Create Budget Plan (V0) >  Confirm > Create/Update Budget Control > Done > New Revision
2. Go to Budget Plan (V1) > Confirm Create/Update Budget Control
3. Go to Budget Control (V1) > Reset Plan
4. it throw error overlaps budget

-----------------------------------------------------------------------------------------------

Budget Plan generate Budget Control and budget control plan (item_ids) generated all AG.
Is it should empty budget control plan and user select KPIs manual or else?
Peek 2021-04-06 13-57

-----------------------------------------------------------------------------------------------

Wizard Generate Budget Control
On Initial KPIs, User should see description, name and budgetable
Selection_001

-----------------------------------------------------------------------------------------------

Initial KPIs 1 AG but it show all AG
Step to error:
1. Create Budget Period > Action > Generate Budget Control Sheet
2. Checked Initial KPIs
3. Selected analytic and Initial KPIs
4. it generated budget control all AG
Peek 2021-04-06 14-19

**Comment by kittiu** [comment]
I think all fixed. But please test more.

**Pull request merged**